### DISCUSSION TITLE
Page not scrollable after showing and hiding dimmer

### DISCUSSION THREAD

**Discussion by indreksiitan** [fomantic/Fomantic-UI#1256] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
Reproducible test case: https://jsfiddle.net/8vhta2b6/

Looks like after the dimmer is shown and hidden, the body element retains "dimmable dimmed" classes, which refrains it from being scrolled. 

The same thing also happens with modals, which I assume use the same underlying dimmer logic. With modals, it also prevents additional modals from being opened.

**Comment by lubber-de** [comment]
That's because you are removing the dimmer while the hide transition still takes place
I commented out the dimmer remove in your onHide callback, it works then
https://jsfiddle.net/hym0ew5j/

The point here is, that dimmer currently doesnt have a `onHidden` callback (after the transition ends) and the onHide callback is called when the hide animation actually starts...i think we should refactor this like in other components, so `onHide` is still being called where it is, but leaves the option to be cancelled by the callback.
An additional `onHidden` instead will definately called when the transition ends.
If you really need to remove the dimmer tag in the current `onHide` callback, use it with a timeout
https://jsfiddle.net/hym0ew5j/1/

**Comment by lubber-de** [comment, marked as answer]
Oh well... no need to refactor as you can already give a callback to the hide behavior 😄 
See https://jsfiddle.net/hym0ew5j/2/

```javascript
$("#progressdimmer").dimmer('hide',function(){
     $("#progressdimmer").remove();
});
```
